lib/puppet/metatype/manager.rb

```diff
@@ -118,10 +118,9 @@ def newtype(name, options = {}, &block)
 
     if klass.is_capability?
       # A capability type is a resource (sort of) so it must be registered with the loader
-      loader = Puppet::Pops::Loaders.loaders.public_environment_loader
-      string_name = name.to_s
-      typed_name = Puppet::Pops::Loader::Loader::TypedName.new(:type, string_name.downcase)
-      loader.set_entry(typed_name, Puppet::Pops::Types::PResourceType.new(string_name.capitalize))
+      loader = Puppet::Pops::Loaders.loaders.resource_type_loader
```

Comment:
I don't get this. Why is it asking for a "resource_type_loader" specifically here instead of an environment or module related loader? Seems like a mis-match in concepts